src/app/icd/ICDMonitoringTable.h

```diff
@@ -59,6 +59,23 @@ struct ICDMonitoringEntry : public PersistentData<kICDMonitoringBufferSize>
     CHIP_ERROR Serialize(TLV::TLVWriter & writer) const override;
     CHIP_ERROR Deserialize(TLV::TLVReader & reader) override;
     void Clear() override;
+    /**
+     * @brief Set the Key object
+     *        This method will create a new keyHandle. The key handle might contain either
+     *        the raw key or a keyID depending on which Crypto implementation is used.
+     *        In any case, to prevent key leakage, one should either call the DeleteKey method
+     *        or save the entry within the ICDMonitoring Table before this object goes out of scope.
+     *
+     *        Calling SetKey() twice on the same object will result in the Key being deleted from
+     *        the keyStore even if the entry was previously saved in the table. One should use a new
+     *        object or manually clear the content of the key handle prior to calling SetKey() again.
```

Comment:
Twice on the same object?  Or twice with the same keydata?  Or something else?

Reply:
Twice on the same object...

Reply:
Here's my suggestion:

1. The API docs should not talk about implementation details (like key handles and what they contain).  They should talk about how to use the API and what its invariants are.
2. "same object" is a bit ambiguous.  Does it mean "same ByteSpan"?  Or "same ICDMonitoringEntry"?
3. What does "manually clear the content of the key handle" mean?  Again, API dos should talk about APIs....
4. Why does calling SetKey twice on the same object fail catastrophically, exactly?  Can we just ... have it not do that?

Reply:
Rephrase the comments to explains 1 through 4